stories/core/FileUpload.stories.tsx

```diff
@@ -0,0 +1,75 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { SvgUpload } from '@itwin/itwinui-icons-react';
+import { action } from '@storybook/addon-actions';
+import { useState } from '@storybook/client-api';
+import { Story, Meta } from '@storybook/react';
+import React from 'react';
+import { FileUpload, FileUploadProps, Textarea } from '../../src/core';
+
+export default {
+  component: FileUpload,
+  argTypes: {
+    className: { control: { disable: true } },
+    style: { control: { disable: true } },
+    onFileDropped: { control: { disable: true } },
+    children: { control: { disable: true } },
+  },
+  title: 'Core/FileUpload',
+} as Meta<FileUploadProps>;
+
+export const Basic: Story<FileUploadProps> = (args) => {
+  const [files, setFiles] = useState<Array<File>>([]);
+
+  return (
+    <FileUpload
+      content={'Drop file to upload'}
+      {...args}
+      onFileDropped={(files) => {
+        setFiles(Array.from(files));
+        action(`${files.length} files uploaded`)();
+      }}
+    >
+      <Textarea
+        placeholder={'Drag a file here'}
+        rows={1}
+        defaultValue={files.map((f) => f.name)}
```

Comment:
maybe add readonly here, not to edit the name manually

Reply:
btw, sure its defaultValue here? I think it should be just value, cause it changes.

Reply:
`value` and `onChange` are for user inputs, which I don't want to control, so `defaultValue` made sense here

Reply:
User inputs? In storybook user can provide onChange?
I tohught in storybook we give actual user examples, how they would use stuff in their work.
Just by design, defaultValue is used wrong here, IMO.

Reply:
We gave a proper example of `FileUpload`. The question then is about controlled input (`value`/`onChange`) vs uncontrolled (`defaultValue`).

I don't think it's used wrong because the value is not really being changed by the user directly.

Reply:
Well, from what I tried, value changes when I drop the file in there...
Again, you are missusing here the defaultValue. https://reactjs.org/docs/uncontrolled-components.html
If you want to have it uncontrolled, you need to give defaultValue a static value: "no files were uploaded" or something like that.

Reply:
The value changes but that's not user input. I would even call it "initial state".

But I think the confusion here might actually come from using the Textarea wrong (we are using it for both user input and for showing filename).

So maybe I can make the textarea readonly, or I can put the filename outside the textarea.

Reply:
What do you mean it is not an user input? So if I just dragged the files in there, the names which appeared are not my input??
The only issue in here is, that you should use value instead of defaultValue, thats all...

Reply:
oh, and readonly would be useful too in this context

Reply:
readonly textarea looks like it's disabled ([work item](https://github.com/iTwin/iTwinUI/projects/2#card-60322339))

so I've changed story to use `LabeledInput`